stdlib/test/builtin/test_str.mojo

```diff
@@ -16,7 +16,10 @@ from testing import assert_equal
 
 
 def test_str_none():
-    assert_equal(str(None), "None")
+    # TODO(#3393): Change to str(None) when MLIR types do not confuse overload resolution.
```

Comment:
**Question** Is this still valid since we have `NoneType` in the `stdlib` now?  We could give the conformance of `RepresentableCollectionElement` to it and the previous code may "just work"?